back-end/middlewares/verifyToken.js

```diff
@@ -0,0 +1,20 @@
+const admin = require("firebase-admin");
+
+const verifyToken = async (req, res, next) => {
+  try {
+    // don't need to check for token when user is trying to register
+    if (!req.url?.startsWith("/auth")) {
+      const token = req.headers.authorization;
+      const decodedToken = await admin.auth().verifyIdToken(token);
+      req.user = decodedToken; //now we can use req.user to get the user object
```

Comment:
Maybe make this part of the request body, instead of a new object?

Reply:
This is not an issue, once logged in Firebase automatically puts user in the headers of all the requests. Leave as is